presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -1046,28 +1055,167 @@ public QueryInfo updateQueryInfo(Optional<StageInfo> stageInfo)
      * Remove large objects from the query info object graph, e.g : plan, stats, stage summaries, failed attempts
      * Used when pruning expired queries from the state machine
      */
-    public void pruneQueryInfo()
+    public void pruneQueryInfo(PruneLevel level)
     {
         Optional<QueryInfo> finalInfo = finalQueryInfo.get();
         if (!finalInfo.isPresent() || !finalInfo.get().getOutputStage().isPresent()) {
             return;
         }
 
         QueryInfo queryInfo = finalInfo.get();
+        QueryInfo prunedQueryInfo;
+        switch (level) {
+            case FINISHED:
```

Comment:
PruneLevel has values `FINISHED` and `EXPIRED`. They don't sound like levels

Reply:
I updated the enum name to `PruneEvent`